packages/runtime/id-compressor/src/idCompressor.ts

```diff
@@ -697,7 +697,9 @@ export class IdCompressor implements IIdCompressor, IIdCompressorCore {
 		let baseFinalId = 0;
 		for (let i = 0; i < clusterCount; i++) {
 			const sessionIndex = readNumber(index);
-			const session = sessions[sessionIndex + sessionOffset][1];
+			// TODO why are we non null asserting here?
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			const session = sessions[sessionIndex + sessionOffset]![1];
```

Comment:
Why not just actually `assert(session !== undefined, "Expected a session")`?

Ditto for the other places in this PR where the `!` was introduced.